flink-java/src/main/java/org/apache/flink/api/java/operators/GroupReduceOperator.java

```diff
@@ -156,8 +162,8 @@ public boolean isCombinable() {
 	
 	public GroupReduceOperator<IN, OUT> setCombinable(boolean combinable) {
 		// sanity check that the function is a subclass of the combine interface
-		if (combinable && !(function instanceof GroupCombineFunction)) {
-			throw new IllegalArgumentException("The function does not implement the combine interface.");
+		if (combinable && !(function instanceof GroupCombineFunction || function instanceof CombineFunction)) {
```

Comment:
If we change the `checkCombinability` method to return a boolean instead of setting `this.combinable` directly, we can also use it here.